v2/gocli/go_binary_test.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gocli_test
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-licenses/v2/gocli"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestListModulesInGoBinary(t *testing.T) {
+	var tests = []struct {
+		workdir     string
+		mainModule  string
+		packagePath string
+		modules     []string
+	}{
+		{
+			workdir:     "../tests/modules/hello01",
+			mainModule:  "github.com/google/go-licenses/v2/tests/modules/hello01",
+			packagePath: "github.com/google/go-licenses/v2/tests/modules/hello01",
+			modules:     []string{},
+		},
+		{
+			workdir:     "../tests/modules/cli02",
+			mainModule:  "github.com/google/go-licenses/v2/tests/modules/cli02",
+			packagePath: "github.com/google/go-licenses/v2/tests/modules/cli02",
+			modules: []string{
+				"github.com/fsnotify/fsnotify",
+				"github.com/hashicorp/hcl",
+				"github.com/magiconair/properties",
+				"github.com/mitchellh/go-homedir",
+				"github.com/mitchellh/mapstructure",
+				"github.com/pelletier/go-toml",
+				"github.com/spf13/afero",
+				"github.com/spf13/cast",
+				"github.com/spf13/cobra",
+				"github.com/spf13/jwalterweatherman",
+				"github.com/spf13/pflag",
+				"github.com/spf13/viper",
+				"github.com/subosito/gotenv",
+				"golang.org/x/sys",
+				"golang.org/x/text",
+				"gopkg.in/ini.v1",
+				"gopkg.in/yaml.v2",
+			},
+		},
+	}
+	originalWorkDir, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, tc := range tests {
+		t.Run(tc.workdir, func(t *testing.T) {
+			os.Chdir(filepath.Join(originalWorkDir, tc.workdir))
+			// This outputs the built binary as name "main".
+			binaryName := "main"
+			cmd := exec.Command("go", "build", "-o", binaryName)
```

Comment:
I noticed when playing around with the tests that the test is not cleaning up after itself. Lets output binaries to a temp directory with [ioutil.TempDir](https://pkg.go.dev/io/ioutil#TempDir) and defer cleanup.

Reply:
done